site/docs/api/serialization.md

```diff
@@ -315,9 +316,58 @@ hazelcast:
         "class-name": "com.hazelcast.jet.examples.PersonSerializer"
 ```
 
-All the classes - data types, serializers & hooks - should be present
-on the cluster classpath, ideally in server's `lib` directory packaged as
-a jar file. Moreover, used type ids have to be unique across all
-serializers. Despite those limitations cluster level serializers offer
-full support for IMDG [sources and sinks](sources-sinks.md) - in particular
-the possibility to query & update `Map`s as well as read from `EventJournal`.
+Put the JAR containing the serializer hook and related classes in the
+`<jet_home>/lib` directory. Make sure that ecah registered serializer
+has a unique type ID.
+
+The advantage of a cluster-level serializer is that it is supported in
+all Hazelcast Jet features.
+
+## 3rd-Party Serialization Support
+
+### Google Protocol Buffers
+
+Since the classes generated by Google Protocol Buffers (Protobuf)
+already implement `java.io.Serializable`, Hazelcast Jet automatically
+supports them without a custom serializer. However, for best performance
+we encourage registering a Protobuf-specific serializer. There is a Jet
+extension module that simplifies this for Protobuf version 3.
+
+Add the extension as a dependency to your Jet job's Maven project:
+
+```xml
+<dependency>
+    <groupId>com.hazelcast.jet</groupId>
+    <artifactId>hazelcast-jet-protobuf</artifactId>
+    <classifier>jar-with-dependencies</classifier>
```

Comment:
since it's reworded, we don't need to depend on the fat JAR here, and we should also include the equivalent gradle snippet as the first option (this is the convention throughout the documentation)

Reply:
We actually want to support both approaches, with the extension installed on the Jet cluster and used just inside a job. We should update the docs in that direction.